### DISCUSSION TITLE
Best way to integrate with an external rAF loop e.g. gsap.ticker()

### DISCUSSION THREAD

**Discussion by droplab** [idiotWu/smooth-scrollbar#439] [Q&A]
This would improve performance and sync better with gsap animations etc.. Wondering if this could be done in a plugin although it would seem doubtful as this is would be pretty low level change I would think.

**Comment by dolphin-wood** [comment]
Well, it's actually possible with the plugin system:

```ts
class RAFPlugin extends ScrollbarPlugin {
  static pluginName = 'raf';

  override onRender() {
    // `onRender()` will be called in every render loop (rAF loops)
  }
}
```

**Reply by andrewleek-droplab** [reply]
I believe @droplab was asking how to drive the internal smooth-scrollbar render/ticker with the gsap ticker, not drive the gsap ticker via the internal smooth-scrollbar rAF loop.

I have forked your repo and done just this:

https://github.com/andrewleek-droplab/smooth-scrollbar/commit/122832e4e8c680d484191776c415c723dd3c22db

I would enjoy knowing your thoughts on this, and if you are interested i can make a PR.
